Author: Matteo Bini <matteobin@tiepi.it>
Description: Add environment variables CPPFLAGS and CFLAGS to compiler
 options and LDFLAGS to linker options,
 to fix hardening Lintian warning (bindnow).
Forwarded: not-needed
Last-Update: 2026-04-29
--- a/CMakeLists.txt	2026-02-14 16:57:38.000000000 +0100
+++ b/CMakeLists.txt	2026-04-29 21:20:24.659445562 +0200
@@ -104,9 +104,12 @@ endif()
 
 add_executable(cdemu-daemon ${cdemu-daemon_SOURCES})
 
+target_compile_options(cdemu-daemon PRIVATE "SHELL:$ENV{CPPFLAGS}")
+target_compile_options(cdemu-daemon PRIVATE "SHELL:$ENV{CFLAGS}")
 target_link_libraries(cdemu-daemon PRIVATE PkgConfig::GLIB)
 target_link_libraries(cdemu-daemon PRIVATE PkgConfig::LIBMIRAGE)
 target_link_libraries(cdemu-daemon PRIVATE PkgConfig::AO)
+target_link_options(cdemu-daemon PRIVATE "SHELL:$ENV{LDFLAGS}")
 
 if(ENABLE_LOGIND_SLEEP_HANDLER AND GLIB_glib-2.0_VERSION VERSION_LESS "2.74")
     target_link_libraries(cdemu-daemon PRIVATE PkgConfig::GIO_UNIX)
